### ISSUE TITLE
Notifications not sticky

### ISSUE THREAD

**Issue by Tadly** [crmarsh/discord-linux-bugs#87]
Currently, when receiving notifications (PMs, Mentions, etc.), they show up for ~5 sec. and than disappear again.
In case of DEs with a notification center (Gnome3, Budgie etc.) it'd be nice for those notifications to be sticky so I know someone pinged me while I was away.

**Comment by SwooshyCueb** [comment]
IMO this should be user-configurable. I do feel that the notifications do not stay long enough, but I wouldn't want them stickied.

**Comment by crmarsh** [comment]
Well, currently we just use the system notifications provided via libnotify, and my understanding is that each window manager or whatever can implement them whoever they like... which is chaos, for sure, but maybe The One True Linux Way.

For Windows less than 10 there is no system notification system, so we built our own version. I think I might expose optionally using that via a config option. Otherwise, I don't really have a lot of control over how/how long they are displayed, you'd have to take it up with the maker of whatever system you are using.

**Comment by zeeshan595** [comment]
Thunderbird Mail has a number. Would be nice if discord had something similar. I added a screenshot for more clear view :)
![screenshot from 2016-09-16 18-44-10](https://cloud.githubusercontent.com/assets/6005448/18595756/06412050-7c3e-11e6-8949-efa6ff061a7f.png)

**Comment by Tadly** [comment]
@crmarsh sorry for my late response.
One question. When you say...

>currently we just use the system notifications provided via libnotify

...do you mean that **you** implemented notification handling yourself or do you actually mean electron?

I assume you mean electron.
I did a little bit of digging and as it seams, electron and notifications are pretty bare bones right now at least in terms of linux.
If you guys use electrons notification implementation, this issue might actually shift to electron than.

**Comment by greenbigfrog** [comment]
I think that actually doesn't matter, since electron (if it has a notification system) also uses libnotify under the hood.
(At least on all setups where libnotify is my primary notification system, I've been getting electron notifications via libnotify, since else they'd look like the chrome/chromium alerts)

**Comment by Tadly** [comment]
@greenbigfrog well yeah, electron uses libnotify.
The point I was trying to make is:

If discord implements there own notification-posting (which I'm unsure of due to the way crmarsh phrased his sentence) for all platforms, this enhancement belongs to discord.
If they just use whatever electron provides, than electron should improve it.

I'm almost certain they use electron's implementation though :)

**Comment by greenbigfrog** [comment]
Yes. They are `currently [...][using] the system notifications provided via libnotify`
So just configure your notification server (which gets a "message" from libnotify) for your needs. Discord can't directly access your notification server/service, without implementing the correct stuff for all servers [ref](https://wiki.archlinux.org/index.php/Desktop_notifications)
(At least that's how it would work, if I understand the path of notifications correct)

**Comment by crmarsh** [comment]
Sorry for the confusing way I put it, yeah, we are using the notifications built into electron, which in turn uses libnotify.

**Comment by Tadly** [comment]
Alright.
I finally found where the notification stuff is implemented. It's in the [Brightray](https://github.com/electron/brightray) lib and it's just a wrapper (as expected).

After testing other electron apps (slack for example) the Notification behavior is just as I would expect it to be. It notifies and it stays.

As this is a "closed source" project, I can't really help investigating but I assume you dismiss notifications automatically at some point?

**Comment by crmarsh** [comment]
I think I get what you're talking about now, I dunno why it didn't click for me at first. I took a look at the Javascript, and yeah, we're dismissing the notifications after 3 seconds. I suppose we could add a setting for that. As a default I prefer what we're doing to what Slack does, but I can see where people might want it the other way.

**Comment by Tadly** [comment]
Hehe, as long as it's clear now :)

I mean, notifications aren't as simple as just posting them. They are only really useful if properly implemented (which goes a long way sometimes).

For now, what you proposed should be fine.
Might come back to it after I was able to give it some testing though. :)

**Comment by SwooshyCueb** [comment]
Unrelated to Linux version, but I figured this might be the best place to mention it.
I hate hate hate hate hate Windows 10's notification system. Do you think you might be able to suggest to the Windows devs that an option be included to use the builtin notification system?

**Comment by crmarsh** [comment]
Well, I'm also one of the windows devs, so... noted.
